web-frontend/src/pages/TransferCoins/SendCoins/Send.tsx

```diff
@@ -23,56 +18,33 @@ function Send(props: SendProps) {
 
   let presetTo = query.get('to');
   let presetAmount = query.get('amount') ?? '';
-  const unformattedBalance = account?.candidateBalance ?? '0';
+  const unformattedBalance = account?.candidateBalance ?? '0.00';
   const balance = parseInt(unformattedBalance);
 
   const [amount, setAmount] = useState<string>(presetAmount);
-  const [fees, setFees] = useState<number>(1000);
+  const [fees, setFees] = useState<string>('1000');
   const [recipient, setRecipient] = useState<string>(presetTo ?? '');
   const [valid, setValid] = useState<boolean>(false);
   const [error, setError] = useState<object | null>(null);
   const [modal, setModal] = useState<boolean>(false);
   const [modalAccounts, setModalAccounts] = useState<boolean>(false);
+  const [errorAdvanced, setErrorAdvanced] = useState<object | null>(null);
 
-  function validate(e: FormEvent<HTMLFormElement>) {
-    const formObject = parseForm(e);
-    const { amount, recipient, fees } = formObject;
-    const amountNum = reverseFormatStandard(amount);
-
-    if (amountNum > balance) {
-      setError({ amount: Intl.t('errors.send.amount-to-high') });
-      return false;
-    }
-    if (amountNum <= 0) {
-      setError({ amount: Intl.t('errors.send.amount-to-low') });
-      return false;
-    }
-    if (Number.isNaN(amountNum)) {
-      setError({ amount: Intl.t('errors.send.invalid-amount') });
-      return false;
-    }
-    if (Number.isNaN(fees)) {
-      setError({ fees: Intl.t('errors.send.invalid-amount') });
-      return false;
-    }
-    if (!recipient) {
-      setError({ recipient: Intl.t('errors.send.no-recipient') });
-      return false;
-    }
-    if (!checkRecipientFormat(recipient)) {
-      setError({ recipient: Intl.t('errors.send.invalid-recipient') });
-      return false;
-    }
-    setError(null);
-    return true;
-  }
   function handleSubmit(e: FormEvent<HTMLFormElement>) {
     e.preventDefault();
-    if (!validate(e)) return;
-    else setValid(!valid);
+    const formObject = parseForm(e);
+    const { amount, recipient } = formObject;
+    const errors = validateInputs(
+      amount,
+      recipient,
+      'recipient',
+      unformattedBalance,
+    );
+    setError(errors);
+    if (errors !== null) return;
+    setValid(!valid);
```

Comment:
why `valid = !valid`?

why not `valid = true`

Reply:
I don't really get why yet but it didn't worked when we tried ( cc @pivilartisant  )

Reply:
either works